### ISSUE TITLE
problem when trying to run galaxy.yml playbook

### ISSUE THREAD

**Issue by kpoterlowicz** [ARTbio/GalaxyKickStart#272]
Hi,
I was trying to follow https://artbio.github.io/GalaxyKickStart/getting_started/
to setup galaxy on aws using ansible and when I run the following

ansible-playbook --inventory-file inventory_files/cloud galaxy.yml

on my host, I got the following error ( I replaced my details with *****)

fatal: [<**************>]: UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug1: Control socket \"/home/******/.ansible/cp/ansible-ssh-<**************>-22-ubuntu\" does not exist\r\nssh: Could not resolve hostname <**************>: Name or service not known\r\n", 
    "unreachable": true
}

Kind regards,
Krzys

**Comment by drosofff** [comment]
Hi,

I ping @afgane for the cloud role.

In the mean time, I used to proceed with aws exactly as any remote machine, with a inventory_files/galaxykickstart like this:

```
[galaxykickstart]
xxx.yyy.zzz.ttt ansible_ssh_user="ubuntu" ansible_ssh_private_key_file="~/.ssh/id_rsa"
[aws]
xxx.yyy.zzz.ttt ansible_ssh_user="ubuntu" ansible_ssh_private_key_file="~/.ssh/id_rsa"

```
where xxx.yyy.zzz.ttt stands for the aws external ip address, and ubuntu is generally the default user with sudo rights in ubuntu images available in aws.

You can replace [galaxykickstart] by any inventory, but keep the [aws] part (with the same line) for NAT maskerading and ftp functionalities.

Last, you can also ssh to your aws instance and replace then xxx.yyy.zzz.ttt by localhost

Hope it helps

Chris

**Comment by kpoterlowicz** [comment]
Hi Chris 
Thank you for replying I tried with inventory_files/galaxykickstart as you specified and it failed.
When I added -vvvv flag I got the following response:

*****@*****-HP-Z620-Workstation:~/ansible/GalaxyKickStart$ ansible-playbook -vvvv --inventory-file inventory_files/galaxy-kickstart galaxy.yml
Using /home/*****/ansible/GalaxyKickStart/ansible.cfg as config file
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/system_users.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/ubuntu/packages.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/ubuntu/rabbitmq.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/ubuntu/docker.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/debian/packages.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-os/tasks/debian/docker.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/cloud_setup/tasks/cleanup.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/miniconda-role/tasks/environment.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/clone.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/virtualenv.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/download.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/virtualenv.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/static_setup.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/dependencies.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/virtualenv.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/mutable_setup.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/database.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy/tasks/errordocs.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/galaxy_root.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/galaxy_metrics.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/uwsgi.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/nginx.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/letsencrypt.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/proftpd.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/slurm.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/pbs.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/k8.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/condor.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/cvmfs_client.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/supervisor.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/ie_proxy.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/galaxy_scripts.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/rabbitmq.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-extras/tasks/startup.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxy.movedata/tasks/import.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxy.movedata/tasks/check_psql_connection.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxy.movedata/tasks/export.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxy.movedata/tasks/check_psql_connection.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.trackster/tasks/setup_trackster.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/bootstrap_user.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/tools.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/workflows.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/bootstrap_user.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/handlers/restart_galaxy.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/restart_galaxy_docker.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/restart_galaxy_run_sh.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/galaxyprojectdotorg.galaxy-tools/tasks/restart_galaxy_supervisor.yml
statically included: /home/*****/ansible/GalaxyKickStart/roles/handlers/galaxy.yml
Loading callback plugin default of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/__init__.pyc

PLAYBOOK: galaxy.yml ***********************************************************
1 plays in galaxy.yml

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
Using module file /usr/local/lib/python2.7/dist-packages/ansible/modules/core/system/setup.py
<xxx.yyy.zzz.ttt> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<xxx.yyy.zzz.ttt> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/home/*****/.ssh/20171122v2_galaxy.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/home/*****/.ansible/cp/ansible-ssh-%h-%p-%r xxx.yyy.zzz.ttt '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1511456330.92-27112760203207 `" && echo ansible-tmp-1511456330.92-27112760203207="` echo $HOME/.ansible/tmp/ansible-tmp-1511456330.92-27112760203207 `" ) && sleep 0'"'"''
fatal: [xxx.yyy.zzz.ttt]: UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug1: Control socket \"/home/*****/.ansible/cp/ansible-ssh-xxx.yyy.zzz.ttt-22-ubuntu\" does not exist\r\ndebug2: ssh_connect: needpriv 0\r\ndebug1: Connecting to xxx.yyy.zzz.ttt [xxx.yyy.zzz.ttt] port 22.\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug1: fd 3 clearing O_NONBLOCK\r\ndebug1: Connection established.\r\ndebug3: timeout: 9981 ms remain after connect\r\ndebug3: Incorrect RSA1 identifier\r\ndebug3: Could not load \"/home/*****/.ssh/20171122v2_galaxy.pem\" as a RSA1 public key\r\ndebug1: identity file /home/*****/.ssh/20171122v2_galaxy.pem type -1\r\ndebug1: identity file /home/*****/.ssh/20171122v2_galaxy.pem-cert type -1\r\ndebug1: Enabling compatibility mode for protocol 2.0\r\ndebug1: Local version string SSH-2.0-OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8\r\ndebug1: Remote protocol version 2.0, remote software version OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8\r\ndebug1: match: OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8 pat OpenSSH_6.6.1* compat 0x04000000\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug3: load_hostkeys: loading entries for host \"xxx.yyy.zzz.ttt\" from file \"/home/*****/.ssh/known_hosts\"\r\ndebug3: load_hostkeys: found key type ECDSA in file /home/*****/.ssh/known_hosts:7\r\ndebug3: load_hostkeys: loaded 1 keys\r\ndebug3: order_hostkeyalgs: prefer hostkeyalgs: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521\r\ndebug1: SSH2_MSG_KEXINIT sent\r\ndebug1: SSH2_MSG_KEXINIT received\r\ndebug2: kex_parse_kexinit: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1\r\ndebug2: kex_parse_kexinit: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-dss-cert-v01@openssh.com,ssh-rsa-cert-v00@openssh.com,ssh-dss-cert-v00@openssh.com,ssh-ed25519,ssh-rsa,ssh-dss\r\ndebug2: kex_parse_kexinit: aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-gcm@openssh.com,aes256-gcm@openssh.com,chacha20-poly1305@openssh.com,aes128-cbc,3des-cbc,blowfish-cbc,cast128-cbc,aes192-cbc,aes256-cbc,arcfour,rijndael-cbc@lysator.liu.se\r\ndebug2: kex_parse_kexinit: aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-gcm@openssh.com,aes256-gcm@openssh.com,chacha20-poly1305@openssh.com,aes128-cbc,3des-cbc,blowfish-cbc,cast128-cbc,aes192-cbc,aes256-cbc,arcfour,rijndael-cbc@lysator.liu.se\r\ndebug2: kex_parse_kexinit: hmac-md5-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-ripemd160-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-96-etm@openssh.com,hmac-md5,hmac-sha1,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-ripemd160,hmac-ripemd160@openssh.com,hmac-sha1-96,hmac-md5-96\r\ndebug2: kex_parse_kexinit: hmac-md5-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-ripemd160-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-96-etm@openssh.com,hmac-md5,hmac-sha1,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-ripemd160,hmac-ripemd160@openssh.com,hmac-sha1-96,hmac-md5-96\r\ndebug2: kex_parse_kexinit: zlib@openssh.com,zlib,none\r\ndebug2: kex_parse_kexinit: zlib@openssh.com,zlib,none\r\ndebug2: kex_parse_kexinit: \r\ndebug2: kex_parse_kexinit: \r\ndebug2: kex_parse_kexinit: first_kex_follows 0 \r\ndebug2: kex_parse_kexinit: reserved 0 \r\ndebug2: kex_parse_kexinit: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1\r\ndebug2: kex_parse_kexinit: ssh-rsa,ssh-dss,ecdsa-sha2-nistp256,ssh-ed25519\r\ndebug2: kex_parse_kexinit: aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-gcm@openssh.com,aes256-gcm@openssh.com,chacha20-poly1305@openssh.com,aes128-cbc,3des-cbc,blowfish-cbc,cast128-cbc,aes192-cbc,aes256-cbc,arcfour,rijndael-cbc@lysator.liu.se\r\ndebug2: kex_parse_kexinit: aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-gcm@openssh.com,aes256-gcm@openssh.com,chacha20-poly1305@openssh.com,aes128-cbc,3des-cbc,blowfish-cbc,cast128-cbc,aes192-cbc,aes256-cbc,arcfour,rijndael-cbc@lysator.liu.se\r\ndebug2: kex_parse_kexinit: hmac-md5-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-ripemd160-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-96-etm@openssh.com,hmac-md5,hmac-sha1,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-ripemd160,hmac-ripemd160@openssh.com,hmac-sha1-96,hmac-md5-96\r\ndebug2: kex_parse_kexinit: hmac-md5-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-ripemd160-etm@openssh.com,hmac-sha1-96-etm@openssh.com,hmac-md5-96-etm@openssh.com,hmac-md5,hmac-sha1,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-ripemd160,hmac-ripemd160@openssh.com,hmac-sha1-96,hmac-md5-96\r\ndebug2: kex_parse_kexinit: none,zlib@openssh.com\r\ndebug2: kex_parse_kexinit: none,zlib@openssh.com\r\ndebug2: kex_parse_kexinit: \r\ndebug2: kex_parse_kexinit: \r\ndebug2: kex_parse_kexinit: first_kex_follows 0 \r\ndebug2: kex_parse_kexinit: reserved 0 \r\ndebug2: mac_setup: setup hmac-md5-etm@openssh.com\r\ndebug1: kex: server->client aes128-ctr hmac-md5-etm@openssh.com zlib@openssh.com\r\ndebug2: mac_setup: setup hmac-md5-etm@openssh.com\r\ndebug1: kex: client->server aes128-ctr hmac-md5-etm@openssh.com zlib@openssh.com\r\ndebug1: sending SSH2_MSG_KEX_ECDH_INIT\r\ndebug1: expecting SSH2_MSG_KEX_ECDH_REPLY\r\ndebug1: Server host key: ECDSA b9:32:08:68:07:8a:e8:98:7a:e5:45:e0:9e:ab:29:31\r\ndebug3: load_hostkeys: loading entries for host \"xxx.yyy.zzz.ttt\" from file \"/home/*****/.ssh/known_hosts\"\r\ndebug3: load_hostkeys: found key type ECDSA in file /home/*****/.ssh/known_hosts:7\r\ndebug3: load_hostkeys: loaded 1 keys\r\ndebug1: Host 'xxx.yyy.zzz.ttt' is known and matches the ECDSA host key.\r\ndebug1: Found key in /home/*****/.ssh/known_hosts:7\r\ndebug1: ssh_ecdsa_verify: signature correct\r\ndebug2: kex_derive_keys\r\ndebug2: set_newkeys: mode 1\r\ndebug1: SSH2_MSG_NEWKEYS sent\r\ndebug1: expecting SSH2_MSG_NEWKEYS\r\ndebug2: set_newkeys: mode 0\r\ndebug1: SSH2_MSG_NEWKEYS received\r\ndebug1: SSH2_MSG_SERVICE_REQUEST sent\r\ndebug2: service_accept: ssh-userauth\r\ndebug1: SSH2_MSG_SERVICE_ACCEPT received\r\ndebug2: key: k.murat1@bradford.ac.uk (0x55774ed1a640),\r\ndebug2: key: k.poterlowicz1@bradford.ac.uk (0x55774ed1f690),\r\ndebug2: key: /home/*****/.ssh/20171122v2_galaxy.pem ((nil)), explicit\r\ndebug1: Authentications that can continue: publickey\r\ndebug3: start over, passed a different list publickey\r\ndebug3: preferred gssapi-with-mic,gssapi-keyex,hostbased,publickey\r\ndebug3: authmethod_lookup publickey\r\ndebug3: remaining preferred: ,gssapi-keyex,hostbased,publickey\r\ndebug3: authmethod_is_enabled publickey\r\ndebug1: Next authentication method: publickey\r\ndebug1: Offering RSA public key: k.murat1@bradford.ac.uk\r\ndebug3: send_pubkey_test\r\ndebug2: we sent a publickey packet, wait for reply\r\ndebug1: Authentications that can continue: publickey\r\ndebug1: Offering RSA public key: k.poterlowicz1@bradford.ac.uk\r\ndebug3: send_pubkey_test\r\ndebug2: we sent a publickey packet, wait for reply\r\ndebug1: Authentications that can continue: publickey\r\ndebug1: Trying private key: /home/*****/.ssh/20171122v2_galaxy.pem\r\ndebug1: key_parse_private2: missing begin marker\r\ndebug1: read PEM private key done: type RSA\r\ndebug3: sign_and_send_pubkey: RSA 2c:35:99:1a:f2:a4:6f:e9:f8:1b:85:12:d9:fc:7b:77\r\ndebug2: we sent a publickey packet, wait for reply\r\ndebug1: Authentications that can continue: publickey\r\ndebug2: we did not send a packet, disable method\r\ndebug1: No more authentication methods to try.\r\nPermission denied (publickey).\r\n", 
    "unreachable": true
}
	to retry, use: --limit @/home/*****/ansible/GalaxyKickStart/galaxy.retry

PLAY RECAP *********************************************************************
xxx.yyy.zzz.ttt                : ok=0    changed=0    unreachable=1    failed=0 

Many thanks,
Krzys

**Comment by drosofff** [comment]
Can I see the content of your inventory file please ? (you can mask the read IP address if you wish)

You have a problem of ssh connection from your local machine. If you can ssh your aws instance, use this exact command line (adapting the path of the ssh key) in your inventory

**Comment by kpoterlowicz** [comment]
Hi,
Thank you very much, that was the problem, fixed my ssh and it works, closes #272
